README.md

```diff
@@ -1,2 +1,145 @@
 # tally-cpp
-C++ tally client
+
+Fast, buffered, hierarchical stats collection in C++.
+
+## Abstract
+
+Tally provides a common interface for emitting metrics, while letting you not worry about the
+velocity of metrics emission.
+
+By default it buffers counters, gauges and histograms at a specified interval, but does not
+buffer timer values.  This is primarily so timer values can have all their values sampled
+if desired and if not they can be sampled as summaries or histograms independently by a reporter.
+
+## Structure
+
+- **Scope**: Keeps track of metrics and their common metadata.
+- **Metrics**: Counters, Gauges, Timers and Histograms.
+- **Reporter**: Implemented by you. Forwards aggregated values from a scope to your metrics
+ingestion pipeline. A default implementation for emitting tagged metrics using Thrift is provided.
+
+### Usage
+
+```c++
+#include <chrono>
+#include <map>
+#include <memory>
+#include <string>
+#include <thread>
+
+#include <m3/reporter.h>
+#include <m3/reporter_builder.h>
+#include <tally/buckets.h>
+#include <tally/scope.h>
+#include <tally/scope_builder.h>
+
+int main(int argc, char** argv) {
+  // Create a reporter that will be used to emit your metrics.
+  auto reporter =
+      tally::m3::ReporterBuilder()
+          .host("127.0.0.1")
+          .port(9052)
+          .common_tags({{"service", "statsdex"}, {"env", "production"}})
+          .Build();
+
+  // Create a scope which serves as a registry for your metrics.
+  auto scope = tally::ScopeBuilder().reporter(reporter).ReportEvery(
```

Comment:
Hm weird to mix a few casing here, could you `s/reporter(..)/Reporter(..)/`.

Also weird to have a builder return a scope on ReportEvery(). Could you require users to call `Build()` as above

Reply:
Sure, in that case I'll just get rid of `ReportEvery` in favor of `Build`. I was following the conventions of the Java client with the two methods.

Reply:
If we just have `Build()` how does one spin up an async report loop?  Launch their own thread (not ideal)?

Reply:
My thought was that the async loop would only be started if the report interval was positive.